pil-analyzer/src/pil_analyzer.rs

```diff
@@ -257,6 +258,64 @@ impl PILAnalyzer {
         }
     }
 
+    /// Verifies that all struct instantiations match their corresponding declarations (existence of field names, completeness)
+    fn struct_fields_check(&self) -> Result<(), Vec<Error>> {
+        let mut errors = vec![];
+
+        self.all_children().for_each(|expr| {
+            if let Expression::StructExpression(sr, struct_expr) = expr {
```

Comment:
You could decrease the indentation here by either using `filter_map` or `for expr in ... { let Expression::StructExpression{ ... } else { continue };` or something else. This is good because when someone reads until here, it is good to know that there will be no `else` clause.